Implement Book Soothsaying for dummies + Profession fixes + Trainer overhaul

FlagFlayer opened the pull request:
## 🍰 Pullrequest
<!-- Describe the Pullrequest. -->
This pull request implements the intended ways to change profession specialisations, as well as generally reworking profession handling to use cmangos implementation.

## Server hosts be warned: merging this PR will cause respecialisation to be broken on old characters

The blizz-like implementation uses hidden "specialisation reputation" (details below) that would not have been rewarded to characters that have previously completed specialisation quests. Once this PR is ready for merge there will be another PR linked here that uses quest completion instead of the reputation to prevent any bugs.

## Changes

### Remove QUEST STATUS NONE update for abandoned professions

* Fixes https://github.com/vmangos/core/issues/1681

* Allows for implementation of Book Soothsaying for dummies

### Implement Book Soothsaying for dummies for Engineering and Leatherworking specialisations

* Fixes https://github.com/vmangos/core/issues/1685
[Added patch 1.10](https://warcraft.wiki.gg/wiki/Patch_1.10.0), before this there was no way to switch specialisations for Engineering and Leatherworking.

> Professions
> Players that unlearn Engineering or Leatherworking may now respecialize if they decide to relearn the professions.

[Video from original WotLK](https://www.youtube.com/watch?v=9ffNV2ZRI7M)

* [Adds relevant gossip to trainer NPCs](https://www.youtube.com/watch?v=u9oOnczxaW8&t=65s)

### Remove profession specialisation ranks in spell_chain

Implied by SkillLineAbility.dbc: There's no previous rank for profession specialisations. They are not part of the profession spell chain.

* Fixes https://github.com/vmangos/core/issues/1686

* Fixes https://github.com/vmangos/core/issues/1683

> From Thottbot By naiad [on 2005/10/27](https://www.wowhead.com/classic/quest=5143/tribal-leatherworking#comments:id=2738463) (Patch 1.8.1)		
> Subject: "225->300 (artisan)"
> there is NO correlation between tribal (or any of the three specs) and getting your artisan (225 to 300 max) leatherworking.  to learn artisan lwing, you will need to go to the master leatherworking trainer.  for the alliance he is in aerie peak, in the inn downstairs.  you must be level 35 and have 200+ lwing.  it will cost you 5 gold.
> 
> for the horde, i believe you have to go to feralas (camp mojache).

> ireburu [on 2006/04/16](https://www.wowhead.com/classic/quest=5141/dragonscale-leatherworking#comments:id=2762152:reply=1184650) (Patch 1.9.4)
> Subject: "actually"
> you don't have to be artisan to get and complete the dragonscale leatherworking quest.

> From Thottbot By ayu129 [on 2006/02/27](https://www.wowhead.com/classic/quest=3641/show-your-work#comments:id=2742849) (Patch 1.9.4)		
> Subject: "255 enginering"
> i beat quest and my max skill says 225 and theres a blue q mark over him head the quest is card renewal and wen i try to renew it it says duplicate idem not found i dont kno how to max ur skill to 300 but if u kno repond plz

[Forum post](https://web.archive.org/web/20060328110216/http://forums.worldofwarcraft.com/thread.aspx?fn=wow-professions&t=146081&p=1&tmp=1#post146081):

> In order to raise your Engineering skill past Expert (225) you must speak with the Master Engineer in Gadetzan and learn Artisan Engineering. Specializing in Goblin or Gnome does NOT let you raise your skill to 300, it simply allows you to access their schemata.

### Implement vanilla mechanics and scripts for Blacksmithing specialisation quests

Blacksmithing quests are unlocked by a gossip option by either Myolor Sunderfury (Alliance) or Krathok Moltenfist (Horde) that gives the player reputation for the hidden "Armorsmithing" or "Weaponsmithing" factions. There is a small gossip script that fires where the respective faction's "guide" points at the relevant NPC that starts the quest for the specialisation you chose and says a short line of text.

> From ThottbotBy dynamite2k [on 2004/10/07](https://www.wowhead.com/classic/npc=11176/krathok-moltenfist#comments:id=2829143) (Patch ?????)		
> Subject: "Gives quests for advanced blacksmithing"
> [Krathok Moltenfist] presents a choice to follow the path of the armorsmith or weaponsmith.
> Can only choose one..

> From AllakhazamBy iamDaMan [on 2006/03/28](https://www.wowhead.com/classic/npc=11178/borgosh-corebender#comments:id=2912151) (Patch 1.9.4)		
> Subject: "got to be done"
> be sure to talk to the guy right next to [Borgosh Corebender] to tell him whether u want to be a weapon smith or an amorsmith

Armorsmithing/Weaponsmithing gossip script:

Example for Horde Weaponsmith: [Video](https://www.youtube.com/watch?v=ymf7WHP2XxY&t=138s)

Example for Horde Armorsmith: [Video](https://www.youtube.com/watch?v=qoHGKYkbrWM&t=789s)

Example for Alliance Armorsmith: [Video](https://youtu.be/MEerghqwVNg)

For Alliance there is additional gossip given by the relevant Blacksmith trainers in Ironforge and Stormwind that guides you toward Myolor to choose a specialisation:

Example for Bengus Deepforge: [Video](https://youtu.be/vEXrfqCRy_w?si=lJyjLpoGvSY1Qn-c&t=1141)

Therum Deepforge: [Relevant Wiki article](https://warcraft.wiki.gg/wiki/Therum_Deepforge#Gossip)

[This gossip remains available](https://youtu.be/uuH8TkDZoH8?si=3RwTJzuEopzNhAGy&t=88) as long as you have the required 200 blacksmithing (and level 40+) but do not have a blacksmithing specialisation.

#### Bengus Deepforge gossip

The wiki article for Therum Deepforge also reveals missing gossip. From looking at the PTR I have concluded this gossip was bugged during vanilla due to being incorrectly displayed at a Blackmsmithing skill of 1:

> "Ye have the look of a Blacksmith to ye. Soot under the nails, steel in yer bones. Aye, very much the blood of a Blacksmith in yer veins. Which means ye've come to me looking to learn a thing or two, well $g lad : lass; grab a hammer. Let's see what I can teach ye."

This gossip would be given before Therum Deepforge can train you. However, currently on the Classic PTR when you have trained blacksmithing but your skill is under 50 he says the following:

![WoWScrnShot_092624_100911](https://github.com/user-attachments/assets/75021774-db60-47cb-a23e-754a1c14c082)

This gossip does not exist in `broadcast_text`, meaning it's a later addition to fix a bug relating to this NPCs gossip, which could only have been due to using this second gossip when he shouldn't have.

Although it's worth mentioning that the [relevant Wiki article](https://warcraft.wiki.gg/wiki/Therum_Deepforge#Gossip) lists the gossip under "Trained" which would imply this behaviour, without other circumstances this would not constitute any sort of proof.

### No vanilla system for changing Blacksmithing specialisations

[An announcement](https://web.archive.org/web/20061109032643/http://www.worldofwarcraft.com/burningcrusade/townhall/professions.html) on new features for TBC all but outright confirms that specialisation change was impossible in vanilla.

### Profession specialisation 'dummy' factions

Different profession specialisation quests reward reputations with hidden factions with the corresponding name (e.g. Completing the quest Tribal Leatherworking rewards reputation with a faction called [Leatherworking - Tribal](https://www.wowhead.com/classic/faction=549/leatherworking-tribal)). This useful for cutting down on the amount of conditions required for the implementation (no need to seperately check horde and alliance quests).

### Pre-1.10 profession specialisation mechanics

Before Patch 1.10, on abandoning and retraining a profession which you specialised in, LearnQuestRewardedSpells would reteach whichever specialisation you had immediately on training the profession.

All relearning methods via gossip (Leatherworking & Engineering via Soothsaying and Blacksmithing & Weaponsmithing sub specialisations via their trainers) were added in 1.10 with the change to LearnQuestRewardedSpells to no longer reteach specialisations immediately on retraining.

[Forum post:](https://web.archive.org/web/20060328110417/http://forums.worldofwarcraft.com/thread.aspx?fn=wow-professions&t=146081&p=28)

> If you specialize in a profession and then drop that profession, resuming it will cause a bug to fire on you. You will be in your old specialty, but will be unable to train in anything new or choose the other specialty. A GM Character Specialist can help this, but it is not a quick turnaround. Before you quit a specialized profession, you should really be sure you don't want to do it anymore.

> If you do quit Engineering, after having specialized, and then return to the profession later, you will still be a member of your chosen specialty. You will be able to renew your membership card and make and learn schematics from your school, with the following exception: You will be unable to learn any specialty schematics introduced since patch 1.4, on 4/19/2005. You may not be able to learn any additional specialty schematics at all, though this is unconfirmed. Though you may be a card carrying member and a 300+ Engineer, you will be told you cannot learn the schematic as you must have the specialization for it. Which you do. And there's the bug.

This PR does not implement the bugged behaviour mentioned above on being unable to learn specialty schematics (mainly as I lack the knowledge on how to implement it). Feel free to add it in the future...

### Deprecate field `trainer_...` fields in `creature template`, implement `creature_trainer` table

Remove old deprecated code and implement  `creature_trainer` table. As per sniffs, all trainer NPCs should have a `gossip_menu_id` of 0 in `creature_template`. The trainer gossip `menu_id` as well as the `trainer_id` should be assigned in `creature_trainer`.

### How2Test
<!-- Give a detailed description how to test your PR and confirm it is working as expected.
- Test1
- Test2
-->
- TBA

### Todo / Checklist
<!-- In case some parts are still missing, important notes, breaking changes and other notable items, list them here. -->
- [x] Write implementation for book "Soothsaying for Dummies"
- [x] Fix automatic quest status none for profession quests
- [x] Remove profession specialisations from profession chain_spell
- [x] Implement blacksmith specialisation gossip and script to unlock quest
- [x] Implement hidden faction reputation for profession quests
- [x] Implement pre-1.10 automatic relearn mechanic
- [x] Correct conditions for Membership Card Renewal
- [ ] Complete deprecation of `trainer_...` fields in `creature_template` and implement `creature_trainer`
- [ ] Test that everything works correctly

FlagFlayer committed:
Create 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Merge branch 'vmangos:development' into soothsaying

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

missing comma

FlagFlayer committed:
Update 20221107191704_world.sql

comma again

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

ratkosrb commented:
Editing broadcast_text is not allowed. That table is bruteforced from classic. The text fields are 100% correct.

ratkosrb commented:
And the male and female texts are per the gender of the NPC which has the gossip, not the player who talks to them.

FlagFlayer commented:
Ah sorry for the confusion on my side, I'll fix it right away

FlagFlayer committed:
Update 20221107191704_world.sql

Remove changes to `broadcast_text`, and add text to NPCs appropriately.

FlagFlayer committed:
Update Player.cpp

Not sure what I'm trying to do, but I hope it works

FlagFlayer committed:
Update 20221107191704_world.sql

I always forget doing the commas

FlagFlayer committed:
Update Player.cpp

FlagFlayer committed:
Update Player.cpp

FlagFlayer committed:
Update Player.cpp

FlagFlayer committed:
Update Player.h

FlagFlayer committed:
Update Player.cpp

FlagFlayer committed:
Update Player.cpp

FlagFlayer commented:
https://github.com/vmangos/core/issues/1683 and https://github.com/vmangos/core/issues/1686 are also related to the vmangos implementation of professions, so ideally those too would also be fixed by importing the cmangos code.

FlagFlayer committed:
Merge branch 'vmangos:development' into soothsaying

FlagFlayer commented:
I finally had the opportuinity to actually test this PR... And it seems it works? I tried abandoning and retaking a profession and it is no longer deleting the quest complete flag. Considering the fact I know basically nothing about C++ I'm impressed by my past self. Need to check for bugs, and test the actual implementation for Soothsaying for Dummies, but this is looking very promising at the moment.

EDIT: I did find 1 minor bug so far, but it's not related to this PR, as it exists in current vmangos. I'll submit a bug report.

FlagFlayer committed:
Fixed mistake in condition.

FlagFlayer committed:
Change condition again

Not sure why this isn't behaving like I'm expecting

FlagFlayer committed:
Update 20221107191704_world.sql

Testing if condtion flag not properly negating the result is the cause of the issue

FlagFlayer committed:
Update 20221107191704_world.sql

Replace negation with flag, did not fix weird behaviour

FlagFlayer committed:
Fix conditions referencing conditions with higher ID

FlagFlayer committed:
Update 20221107191704_world.sql

update comment

FlagFlayer committed:
Fix trainer gossip

FlagFlayer committed:
Merge branch 'vmangos:development' into soothsaying

FlagFlayer commented:
The good news: The conditions work properly now

The bad news: The book is not interactable for some reason

FlagFlayer commented:
~~Other bad news: It seems the profession implimentation has a major bug that causes "trainer_spell" requirement in creature_template to be ignored.~~

Nope. Basic mistake here. The other bad news is actually that the bug regarding retaining of developmental skills still occurs for some reason.

FlagFlayer committed:
Remove "fix" that breaks specialisation training

FlagFlayer committed:
Update Player.cpp

Undo previous changes

FlagFlayer committed:
Update Player.h

Undo previous changes

FlagFlayer committed:
restore trainer spell removal

FlagFlayer committed:
Update 20221107191704_world.sql

Trainer gossip for leatherworking spec trainers

FlagFlayer committed:
Update 20221107191704_world.sql

Add trainer gossip conditions for engineering spec

FlagFlayer committed:
Update Player.h

Restore changes

FlagFlayer committed:
Update Player.cpp

Restore changes

FlagFlayer committed:
Update Player.h

Importing profession implementation pt 1

FlagFlayer committed:
Update Player.h

LearnDefaultSkills()

FlagFlayer committed:
Update Player.h

Undo

FlagFlayer committed:
Update Player.h

FlagFlayer committed:
Import trade skill implementation

FlagFlayer committed:
Update Player.cpp

FlagFlayer committed:
Update Player.cpp

FlagFlayer committed:
Update Player.cpp

FlagFlayer committed:
Update Player.cpp

FlagFlayer committed:
Update Player.cpp

FlagFlayer committed:
Update SkillHandler.cpp

FlagFlayer committed:
Update SkillHandler.cpp

FlagFlayer committed:
Update SkillHandler.cpp

FlagFlayer committed:
Update Player.cpp

FlagFlayer committed:
Update Player.h

FlagFlayer committed:
Merge branch 'development' into soothsaying

FlagFlayer committed:
Update Player.cpp

FlagFlayer committed:
Update ObjectMgr.h

FlagFlayer committed:
Update Player.cpp

FlagFlayer committed:
Update Player.cpp

FlagFlayer committed:
Update ObjectMgr.cpp

FlagFlayer committed:
Update 20221107191704_world.sql

WIP adding blacksmithing changes

FlagFlayer committed:
Update 20221107191704_world.sql

WIP

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

Add basic respec gossip menus

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Merge branch 'vmangos:development' into soothsaying

FlagFlayer committed:
Merge branch 'vmangos:development' into soothsaying

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update SpellEffects.cpp

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer commented:
Finally this PR is at the point where it only needs testing to make sure nothing is broken. If anyone would like to help test, please either let me know, or post your findings here.

FlagFlayer committed:
Merge branch 'vmangos:development' into soothsaying

FlagFlayer committed:
Merge branch 'vmangos:development' into soothsaying

FlagFlayer committed:
Update 20221107191704_world.sql

Fix gossip not showing correctly

FlagFlayer commented:
Can confirm https://github.com/vmangos/core/issues/1683 is not fixed.

It appears that the bug occurs because the game retrains the specialisation and then developmental skills get retrained due to `spell_chain`

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update 20221107191704_world.sql

derp

ratkosrb commented on sql/migrations/20221107191704_world.sql:
```diff
@@ -225,13 +225,13 @@ UPDATE `quest_template` SET `RewSpell` = 20219  WHERE `entry` = 3641; -- Gnomish
+UPDATE `spell_chain` WHERE `spell_id` = 10658 SET `prev_spell` = 3811, `rank` = 4; -- Correct for Elemental leatherworking
+UPDATE `spell_chain` WHERE `spell_id` = 10660 SET `prev_spell` = 3811, `rank` = 4; -- Correct for Tribal leatherworking
+UPDATE `spell_chain` WHERE `spell_id` = 20219 SET `prev_spell` = 4038, `rank` = 4; -- Correct for Gnomish engineering
+UPDATE `spell_chain` WHERE `spell_id` = 20222 SET `prev_spell` = 4038, `rank` = 4; -- Correct for Goblin engineering
```
Why is the WHERE clause before the SET ? Is that even valid ? I've never seen it swapped like this.

FlagFlayer replied:
Yeah obvious mistake on my part

FlagFlayer committed:
Update 20221107191704_world.sql

double derp

FlagFlayer committed:
Update 20221107191704_world.sql

Remove rewspells (not how it's displayed in game)

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Merge branch 'vmangos:development' into soothsaying

FlagFlayer committed:
Update 20221107191704_world.sql

FlagFlayer committed:
Update Player.cpp

Currently not going to work properly, adding this here so I remember to revisit this

FlagFlayer committed:
Update Player.cpp

doesn't getspellrank use spell_id? was this the issue?

FlagFlayer committed:
Update Player.cpp

FlagFlayer committed:
Update Player.h

FlagFlayer committed:
Revert "Update Player.h"

This reverts commit 8e665044811e2e8e0802534f6b56dab8910d389c.

FlagFlayer committed:
Update Player.cpp

FlagFlayer committed:
Update SpellMgr.h

FlagFlayer committed:
Update SpellMgr.cpp

FlagFlayer committed:
Update Player.cpp

This might fix the bug of specialisations still being retaught

FlagFlayer committed:
Update Player.cpp

Apparently this change made you unlearn every single default pattern on abandonment, but didn't change anything about the specialisation spells

FlagFlayer committed:
Update 20221107191704_world.sql

My misimplementation of spell chains might've been the source of the bug actually

FlagFlayer committed:
Revert recent changes

This reverts commit 52ad142d8cf5646ab9f4cad82fe617ab8ed72818.

Revert "Update Player.cpp"

This reverts commit aad47bc2dc1514223010386e80e8889d8981c522.

Revert "Update SpellMgr.cpp"

This reverts commit 57dd3324e385d17f92bea9b69c88e666537ef91d.

FlagFlayer committed:
Update Player.cpp

profession specialisations no longer in spell chain, so they do not have a spell rank larger than 1

FlagFlayer committed:
Update Player.cpp

Attempt to troubleshoot learnquestrewardedspells

FlagFlayer committed:
Update SpellMgr.cpp

I believe the issue with using this function was that profession specialisations do not teaching the skill they are associated with (intended behaviour)

FlagFlayer committed:
Update Player.cpp

Test changes to function

FlagFlayer committed:
Update SpellMgr.cpp

Perhaps did not write this correctly

FlagFlayer committed:
Update Player.cpp

If this doesn't work I'm going to just filter out the spellids for profession quests in spite of how inelegant that looks

FlagFlayer committed:
Update SpellMgr.cpp

Undo change to IsPrimaryProfessionSpell

Update SpellMgr.cpp

FlagFlayer committed:
Update and rename 20221107191704_world.sql to 20240913171704_world.sql

Incorrect NPC flag?

FlagFlayer committed:
Update 20240913171704_world.sql

Fix gossip_script not firing correctly

FlagFlayer committed:
Update 20240913171704_world.sql

update migration id

FlagFlayer committed:
Remove classic-deriven blacksmith respecialisation

FlagFlayer committed:
Update SkillHandler.cpp

Remove irrelevant changes

FlagFlayer committed:
Update ObjectMgr.cpp

Undo irrelevant changes

FlagFlayer committed:
Update ObjectMgr.h

Undo irrelevant changes

FlagFlayer committed:
Update Player.cpp

Remove irrelevant changes

FlagFlayer committed:
Update Player.h

Remove irrelevant changes

FlagFlayer committed:
Update SpellEffects.cpp

Remove irrelevant changes

FlagFlayer committed:
Update SkillHandler.cpp

Remove unnecessary line break

FlagFlayer committed:
Update Player.cpp

Correct if statement

FlagFlayer commented:
PR finally ready for some final tests to make sure everything works correctly.

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

Forgot to remove this

FlagFlayer committed:
Update Player.cpp

FlagFlayer commented:
Converted to draft as there's still a lot of things to implement

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

missing comma

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

ratkosrb commented:
Not a fan of the custom npc_text and gossip_menu ids.

FlagFlayer commented:
I'm not, either, but I don't have sniffs for them. However, in my view it's better to have a core system to professions implemented regardless of whether the backend is 100% accurate aesthetically (I have provided ample proof for correct usage for every custom id I've used).

FlagFlayer commented:
After a quick check I seem to have erroneously added some of the insertions into npc_text. The missing IDs of the missing entries are very easily known since their neighbouring entries are sorted numerically by broadcast text id. The only thing missing is the gossip menu id for any gossip sub menus

FlagFlayer committed:
Update 20240913171704_world.sql

Removed custom npc_text ids

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

Relearning via gossip only added in 1.10

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update Player.cpp

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
cleanup

FlagFlayer committed:
Update 20240913171704_world.sql

further cleanup

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Rework IsTrainerOf

FlagFlayer committed:
Remove deprecated fields from ObjectMgr

FlagFlayer committed:
Update 20240913171704_world.sql

FlagFlayer committed:
Update and rename 20240913171704_world.sql to 20250304163250_world.sql

FlagFlayer committed:
TODO: CharacterCommands

FlagFlayer commented:
This PR is doing too many things at this point. I think I'll split it into separate PRs: One for Soothsaying for Dummies, one for Misc. fixes and one for removing `trainer_...` fields from `creature_template`
